pvlib/iotools/tmy.py

```diff
@@ -519,3 +519,179 @@ def _read_tmy2(string, columns, hdr_columns, fname):
+    present_weather_codes               Present weather code, see [1], chapter 2.9.1.28
+    precipitable_water                  Total precipitable water contained in a column of unit cross section from earth to top of atmosphere, cm
+    aerosol_otpical_depth               The broadband aerosol optical depth per unit of air mass due to extinction by aerosol component of atmosphere, unitless
+    snow_depth                          Snow depth in centimeters on the day indicated, (999 = missing data)
+    days_since_last_snowfall            Number of days since last snowfall (maximum value of 88, where 88 = 88 or greater days; 99 = missing data)
+    albedo                              The ratio of reflected solar irradiance to global horizontal irradiance, unitless
+    liquid_precipitation_depth          The amount of liquid precipitation observed at indicated time for the period indicated in the liquid precipitation quantity field, millimeter
+    liquid_precipitation_quantity       The period of accumulation for the liquid precipitation depth field, hour
+    =============================       ======================================================================================================================================================
+
+    References
+    ----------
+
+    [1] EnergyPlus documentation, Auxiliary Programs
+    https://energyplus.net/documentation.
+    '''
+    
+    if filename is None:
+        try:
+            filename = _interactive_load()
+        except ImportError:
+            raise ImportError('Interactive load failed. Tkinter not supported '
+                              'on this system. Try installing X-Quartz and '
+                              'reloading')
+    
+    if filename.startswith('http'):
+        # Attempts to download online EPW file
+        # See comments above for possible online sources
+        request = Request(filename, headers={'User-Agent': (
+            'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_5) '
+            'AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.87 '
+            'Safari/537.36')})
+        response = urlopen(request)
+        csvdata = io.StringIO(response.read().decode(errors='ignore'))
+    else:
+        # Assume it's accessible via the file system
+        csvdata = open(filename, 'r')
+    
+    # Read line with metadata
+    firstline = csvdata.readline()
+
+    head = ['loc','city', 'state-prov', 'country', 'data_type','WMO_code', 
+            'latitude', 'longitude', 'TZ','altitude']
+    meta = dict(zip(head, firstline.rstrip('\n').split(",")))
+
+    meta['altitude'] = float(meta['altitude'])
+    meta['latitude'] = float(meta['latitude'])
+    meta['longitude'] = float(meta['longitude'])
+    meta['TZ'] = float(meta['TZ'])
+
+    colnames = ['year', 'month', 'day', 'hour', 'minute', 'data_source_unct', 
+                't_drybulb', 't_dewpoint', 'rel_hum', 'atmospheric_pressure',
```

Comment:
see https://pvlib-python.readthedocs.io/en/latest/variables_style_rules.html for temperatures and humidity